src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/history/model/EventData.java

```diff
@@ -0,0 +1,19 @@
+/* Copyright 2023 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.share.history.model;
+
+public interface EventData {
+}
```

Comment:
Thanks for spotting it. 
The idea behind was to somehow organize classes better. We have a multiple classes with the same name and to make it easier to distinguish I thought we could use java types. At the same time promote reusability. Unfortunately, our code base if not so flexible and reorganizing it quite difficult.
For example, when you need to use FlowEventData or even to convert FlowEventData to FlowEventData, it becomes difficult and easy to be confused. If we know that one of those FlowEventData belongs to certain area, we can mark that area with the interface, such as this EventData, then we can use it in code with java typing: one of the classes inherits from this interface, while other is not. 
So, instead of <T> we can use <T extends EventData> and java will not allow to use the wrong type.
In this case, I will remove the interface, but we can continue this discussion and try to find out whether we can use java typing to structure models better.